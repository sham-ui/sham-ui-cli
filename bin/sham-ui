#!/usr/bin/env node

require( 'commander' )
    .version( require( '../package' ).version )
    .usage( '<command> [options]' )
    .command( 'new', 'generate a new project from a template' ).alias( 'n' )
    .command( 'generate', 'generate a new component' ).alias( 'g' )
    .parse( process.argv );