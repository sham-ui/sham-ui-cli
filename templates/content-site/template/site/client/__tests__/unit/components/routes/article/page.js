import { createDI } from 'sham-ui';
import RoutesArticlePage  from '../../../../../src/components/routes/article/page.sfc';
import renderer from 'sham-ui-test-helpers';

it( 'renders correctly', () => {
    const DI = createDI();
    DI.bind( 'title', {
        change() {}
    } );
    DI.bind( 'router', {
        storage: {
            params: {
                page: 2
            },
            url: '',
            addWatcher() {}
        },
        generate: () => '/'
    } );
    DI.bind( 'store', {
        articleBySlug: jest.fn().mockReturnValueOnce(
            Promise.resolve( { meta: {} } )
        )
    } );
    const meta = renderer( RoutesArticlePage, {
        DI
    } );
    expect( meta.toJSON() ).toMatchSnapshot();
} );
