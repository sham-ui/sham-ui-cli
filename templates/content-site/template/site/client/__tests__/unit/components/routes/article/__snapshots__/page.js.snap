// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Object {
  "Options": Object {
    "ID": "component",
    "article": Object {},
    "dataLoaded": false,
    "errors": Array [],
    "isNotFound": false,
  },
  "html": "
<!--0-->
<!--0-->
<!--0-->
<!--0-->
",
}
`;
